src/data/nat/cast.lean

```diff
@@ -99,4 +99,10 @@ by by_cases a ≤ b; simp [h, max]
 @[simp, elim_cast] theorem abs_cast [decidable_linear_ordered_comm_ring α] (a : ℕ) : abs (a : α) = a :=
 abs_of_nonneg (cast_nonneg a)
 
+-- This lemma looks like it's backwards, but it's what's useful in the wild.
+@[simp, squash_cast] lemma cast_with_bot_cast {α} [add_monoid α] [has_one α] :
```

Comment:
Can we confirm this use of `squash_cast`? I think it might be expecting statements of the form `\u \u x = \u x`.

Reply:
@pnmdln ? Clarifying the use of `squash_cast` is on the to-do list. Right now the solution is to ask Paul-Nicolas.

Reply:
This lemma would be a squash_cast lemma if it was written in the the other direction.
I'm trying to find a way to clarify or replace this label, but one of the things it's used for is, well, squashing expressions like \u \u x into \u x. If you put the label on a backward lemma, the tactic can loop forever (here if alpha = nat)

Reply:
Okay, I've removed the squash_cast.